src/tools/OptionsSchema.json

```diff
@@ -0,0 +1,398 @@
+                    "default": false
+                }
+            }
+        },
+        "LaunchConfigurations": {
+            "type": "object",
+            "properties": {
+                "command": {
+                    "type": "string",
+                    "description": "The command to execute for launching the web browser",
+                    "default": "open"
+                },
+                "args": {
+                    "type": "string",
+                    "description": "The arguments to pass to the command to open the browser. Use ${auto-detect-url} to automatically use the address the server is listening to",
+                    "default": "${auto-detect-url}"
+                }
+            }
+        },
+        "LaunchBrowser": {
+            "type": "object",
+            "description": "Describes options to launch a web browser as part of launch",
+            "default": {
+                "enabled": true,
+                "args": "${auto-detect-url}",
+                "windows": {
+                    "command": "cmd.exe",
+                    "args": "/C start ${auto-detect-url}"
+                },
+                "osx": {
+                    "command": "open"
+                },
+                "linux": {
+                    "command": "xdg-open"
+                }
+            },
+            "properties": {
+                "enabled": {
+                    "type": "boolean",
+                    "description": "Whether web browser launch is enabled",
+                    "default": true
+                },
+                "args": {
+                    "type": "string",
+                    "description": "The arguments to pass to the command to open the browser. Use ${auto-detect-url} to automatically use the address the server is listening to",
+                    "default": "${auto-detect-url}"
+                },
+                "osx": {
+                    "$ref": "#/definitions/LaunchConfigurations",
+                    "description": "OSX-specific web launch configuration options",
+                    "default": {
+                        "command": "open"
+                    }
+                },
+                "linux": {
+                    "$ref": "#/definitions/LaunchConfigurations",
+                    "description": "Linux-specific web launch configuration options",
+                    "default": {
+                        "command": "xdg-open"
+                    }
+                },
+                "windows": {
+                    "$ref": "#/definitions/LaunchConfigurations",
+                    "description": "Windows-specific web launch configuration options",
+                    "default": {
+                        "command": "cmd.exe",
+                        "args": "/C start ${auto-detect-url}"
+                    }
+                }
+            }
+        },
+        "LaunchOptions": {
```

Comment:
If this is difficult, feel free to ignore, but it would be great if we had a base class of LaunchOptions/AttachOptions called 'CommonOptions' with all the options that both take.